ts-client/src/amm/index.ts

```diff
@@ -0,0 +1,935 @@
+    return Math.max(this.tokenA.decimals, this.tokenB.decimals);
+  }
+
+  get isStablePool(): boolean {
+    return 'stable' in this.poolState.curveType;
+  }
+
+  /**
+   * It updates the state of the pool
+   */
+  public async updateState() {
+    const poolState = await getPoolState(this.address, this.program);
+    const [vaultA, vaultB] = await Promise.all([
+      getVaultState(poolState.aVault, this.vaultProgram),
+      getVaultState(poolState.bVault, this.vaultProgram),
+    ]);
+    this.vaultA = vaultA;
+    this.vaultB = vaultB;
+
+    // update spl info
+    const accountsBuffer = await getAccountsBuffer({
+      poolState,
+      connection: this.program.provider.connection,
+      vaultA: this.vaultA,
+      vaultB: this.vaultB,
+      apyPda: this.apyPda,
+    });
+    this.accountsInfo = deserializeAccountsBuffer(accountsBuffer);
+
+    if (this.isStablePool) {
+      // update swap curve
+      const { amp, depeg, tokenMultiplier } = poolState.curveType['stable'];
+      this.swapCurve = new StableSwap(
+        amp.toNumber(),
+        tokenMultiplier,
+        depeg,
+        this.depegAccounts,
+        this.accountsInfo.currentTime,
+      );
+    }
+
+    const poolInfo = calculatePoolInfo(
+      this.accountsInfo.currentTime,
+      this.accountsInfo.poolVaultALp,
+      this.accountsInfo.poolVaultBLp,
+      this.accountsInfo.vaultALpSupply,
+      this.accountsInfo.vaultBLpSupply,
+      this.accountsInfo.poolLpSupply,
+      this.accountsInfo.apy,
+      this.swapCurve,
+      this.vaultA,
+      this.vaultB,
+    );
+    this.poolState = { ...poolState, ...poolInfo };
+  }
+
+  /**
+   * It returns the pool token mint.
+   * @returns The poolState.lpMint
+   */
+  public getPoolTokenMint() {
+    return this.poolState.lpMint;
+  }
+
+  /**
+   * It gets the total supply of the LP token
+   * @returns The total supply of the LP token.
+   */
+  public async getLpSupply() {
+    const account = await this.program.provider.connection.getTokenSupply(this.poolState.lpMint);
+    invariant(account.value.amount, ERROR.INVALID_ACCOUNT);
+
+    return new BN(+account.value.amount);
+  }
+
+  /**
+   * Get the user's balance by looking up the account associated with the user's public key
+   * @param {PublicKey} owner - PublicKey - The public key of the user you want to get the balance of
+   * @returns The amount of tokens the user has.
+   */
+  public async getUserBalance(owner: PublicKey) {
+    const account = await Token.getAssociatedTokenAddress(
+      ASSOCIATED_PROGRAM_ID,
+      TOKEN_PROGRAM_ID,
+      this.poolState.lpMint,
+      owner,
+    );
+    if (!account) return new BN(0);
+
+    const parsedAccountInfo = await this.program.provider.connection.getParsedAccountInfo(account);
+    if (!parsedAccountInfo.value) return new BN(0);
+
+    const accountInfoData = (parsedAccountInfo.value!.data as ParsedAccountData).parsed;
+
+    return new BN(Number(accountInfoData.info.tokenAmount.amount));
```

Comment:
Same here, maybe we can declare the type for parsed token account info, and avoid `Number` cast